smac_planner/src/a_star.cpp

```diff
@@ -313,14 +318,28 @@ bool AStarAlgorithm<NodeT>::createPath(
     // 2) Mark Nbest as visited
     current_node->visited();
 
+    // 2.a) Use an analytic expansion (if available) to generate a path
+    // to the goal.
+    // Then check if it is collision free.
+    // TODO(james-ward): Don't always run this check - base it on the
+    // number of iterations and how close to the goal we are
+    analytic_iterations++;
+    if (analytic_iterations > _best_heuristic_node.first) {
```

Comment:
I just did a little testing, I added a print every time we try to expand, and it only tries 1 time in nearly every path I tested with. I think this condition doesn't work - it should be trying even far away, just not as often.

Reply:
Last comment - make sure that this also runs in the very first iteration. If it's a straight or simple path, we can just override the use of A* entirely.

Reply:
I've tweaked it to use the distance rather than heuristic to calculate iterations between analytic expansions. Also made it check on the first iteration for trivial paths.
This will all be much neater when we can use C++17's `constexpr if`! ;)